lib/streamlit/elements/multiselect.py

```diff
@@ -104,6 +104,14 @@ def _check_and_convert_to_indices(
     return [opt.index(value) for value in default_values]
 
 
+def get_default_count(default: Union[List[Any], Any, None]):
+    if default is None:
+        return 0
+    if not isinstance(default, List):
```

Comment:
This isn't quite correct since the type of `default` should match that of the `default` param passed to the `_multiselect` method. That is, it should be `Union[Iterable[Any], Any, None]`, and the function should return the length of any iterable passed into it.

For an example of a bug that exists right now with this implementation, I could pass a tuple of any length into this function and have it return 1.

Reply:
That's definitely a good point. I initially had it to Iterable but when I went to go look at the documentation, it says that it only accepts lists. Maybe if it's not a list, we could throw a StreamlitAPI as an add on instead of using `Union[Iterable[Any], Any, None]`. What do you think?

![Screen Shot 2022-09-16 at 1 18 07 PM](https://user-images.githubusercontent.com/16749069/190725997-701f5683-3c7e-43a5-9474-9390a61c06e4.png)

Reply:
I don't think that'd be a good solution.

It's a breaking change for anyone that happens to be using non-lists as defaults in multiselects (which isn't the biggest deal given what the documentation says we officially support), but more importantly there's no reason default should only be able to take lists vs other types of iterables. The only thing that we want conceptually here is some kind of collection, whether that be a list, tuple, set, or numpy array (admittedly, the last one isn't a particularly sane thing to do).

Reply:
Hm yeah you're right. I swapped it out. Thanks.